aws/PAS-network-environment-NAT.json

```diff
@@ -1026,6 +1069,392 @@
+            "Properties": {
+                "GroupId": {
+                    "Ref": "PTASG"
+                },
+                "IpProtocol": "tcp",
+                "FromPort": 27017,
+                "ToPort": 27017,
+                "DestinationSecurityGroupId": {
+                    "Ref": "PTASG"
+                }
+            }
+        },
+        "PTASGIngress7": {
+            "Type": "AWS::EC2::SecurityGroupIngress",
+            "Properties": {
+                "GroupId": {
+                    "Ref": "PTASG"
+                },
+                "IpProtocol": "tcp",
+                "FromPort": 6514,
+                "ToPort": 6514,
+                "CidrIp": "0.0.0.0/0"
+            }
+        },
+        "PTASGIngress8": {
+            "Type": "AWS::EC2::SecurityGroupIngress",
+            "Properties": {
+                "GroupId": {
+                    "Ref": "PTASG"
+                },
+                "IpProtocol": "tcp",
+                "FromPort": 7514,
+                "ToPort": 7514,
+                "CidrIp": "0.0.0.0/0"
+            }
+        },
+        "PTASGIngress9": {
+            "Type": "AWS::EC2::SecurityGroupIngress",
+            "Properties": {
+                "GroupId": {
+                    "Ref": "PTASG"
+                },
+                "IpProtocol": "tcp",
+                "FromPort": 22,
+                "ToPort": 22,
+                "DestinationSecurityGroupId": {
+                    "Ref": "PTASG"
+                }
+            }
+        },
+        "PTASGIngress10": {
+            "Type": "AWS::EC2::SecurityGroupIngress",
+            "Properties": {
+                "GroupId": {
+                    "Ref": "PTASG"
+                },
+                "IpProtocol": "tcp",
+                "FromPort": 22,
+                "ToPort": 22,
+                "CidrIp": {
+                    "Ref": "AdministrativeAccessCIDR"
+                }
+            }
+        },
+        "PTASGInress12": {
+            "Type": "AWS::EC2::SecurityGroupEgress",
+            "Properties": {
+                "GroupId": {
+                    "Ref": "PTASG"
+                },
+                "IpProtocol": "tcp",
+                "FromPort": 514,
+                "ToPort": 514,
+                "CidrIp": "0.0.0.0/0"
+            }
+        },
+        "PTASGIngress13": {
+            "Type": "AWS::EC2::SecurityGroupEgress",
+            "Properties": {
+                "GroupId": {
+                    "Ref": "PTASG"
+                },
+                "IpProtocol": "udp",
+                "FromPort": 514,
+                "ToPort": 514,
+                "CidrIp": "0.0.0.0/0"
+            }
+        },
+        "PTASGIngress14": {
+            "Type": "AWS::EC2::SecurityGroupEgress",
+            "Properties": {
+                "GroupId": {
+                    "Ref": "PTASG"
+                },
+                "IpProtocol": "tcp",
+                "FromPort": 514,
+                "ToPort": 11514,
```

Comment:
According to AWS documentation: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-security-group-egress.html
**FromPort**: Start of port range for the TCP and UDP protocols, or an ICMP type number. If you specify icmp for the IpProtocol property, you can specify -1 as a wildcard (i.e., any ICMP type number).
**ToPort**: End of port range for the TCP and UDP protocols, or an ICMP code. If you specify icmp for the IpProtocol property, you can specify -1 as a wildcard (i.e., any ICMP code).

Did you really mean to open all the traffic in range of 514-11514 ?